youtube_dl/utils.py

```diff
@@ -5772,3 +5772,150 @@ def clean_podcast_url(url):
+        output 'f34c6dff1f890e75b6b59422dde3b1acb7d3146586'
+        In order to find a corresponding code in cambro.tv/camhub.com scripts
+        do the following:
+        1. Set a breakpoint at kt_start
+        2. Execute in CDT console when triggered:
+        flashvars._video_url = flashvars.video_url;
+        Object.defineProperty(flashvars, 'video_url', {
+            get: function () {
+                return flashvars._video_url;
+            },
+            set: function (value) {
+                debugger;
+                flashvars._video_url = value;
+            }
+        });
+        3. The second break is where the url re-encoding happens
+        """
+        i = KtPlayerHelper._hash_kt_player_lic_code(lic_code)
+        h = orig_hash[0:limit]
+        for k in range(len(h) - 1, -1, -1):
+            l = k
+            for m in range(k, len(i)):
+                l += int(i[m])
+            while l >= len(h):
+                l -= len(h)
+            n = ""
+            for o in range(0, len(h)):
+                if o == k:
+                    n += h[l]
+                elif o == l:
+                    n += h[k]
+                else:
+                    n += h[o]
+            h = n
+        return h + orig_hash[limit:]
+
+    @staticmethod
+    def get_url(webpage):
+        def search(pattern, string, flags=0):
+            mobj = re.search(pattern, string, flags)
+            if mobj:
+                return next(g for g in mobj.groups() if g is not None)
+            return None
+
+        # extract video url
+        license_code = search(r'license_code:\s+\'(.+?)\'', webpage)
+        video_raw_url = search(r'video_url:\s+\'(.+?)\'', webpage)
+        if not license_code or not video_raw_url:
+            return None
+
+        # decode a real video url
+        parts = video_raw_url.split('/')
+        video_pre_parts = []
+
+        # cut some junk at the beginning
+        for i in range(len(parts)):
+            if parts[i].startswith('http'):
+                video_pre_parts = parts[i:]
+        if len(video_pre_parts) < 6:
+            # it is expected to be
+            # http://example.com/get_file/2/1039a5cd2f433e4d41adf41e0afc1773/223000/223101/223101.mp4/
+            # with a hash value as 5th component
+            raise ExtractorError(f'url too short: {video_pre_parts}')
+
+        # convert video hash to a real one
+        orig_hash = video_pre_parts[5]
+        new_hash = KtPlayerHelper.convert_video_hash(license_code, orig_hash)
+        video_pre_parts[5] = new_hash
+        video_url = '/'.join(video_pre_parts)
+
+        return video_url
+
+
+def date_from_ago(ago_str):
+    """Converts strings like '2 months ago' into YYYYMMDD
+    Returns None if fails
+    """
+    uploaded_re = r'(?P<val>\d+)\s+(?P<unit>(?:years?)|(?:months?)|(?:weeks?)' + \
+        r'|(?:days?)|(?:hours?)|(?:minutes?))\s+ago'
+
+    upload_date = re.search(uploaded_re, ago_str)
```

Comment:
Generally, I'd have put the static regex directly as the first argument of `re.search()`.

Reply:
done